lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystem.java

```diff
@@ -119,8 +119,6 @@ public void deleteDirectory(Location location)
     public void deleteFiles(Collection<Location> locations)
             throws IOException
     {
-        locations.forEach(Location::verifyValidFileLocation);
```

Comment:
this should be still be validated for files, right? In other words, can we skip this validation only for directories (or maybe extend the validation to also include dirs)?